design/open/oidc-support.md

```diff
@@ -0,0 +1,28 @@
+## OIDC support in lakeFS
+
+As suggested by @arielshaqed, we started researching the option to implement OIDC support as part of lakeFS.
+
+In such a solution, the person who configures lakeFS will supply the parameters required for implementing OIDC, including the provider's domain, the client ID and secret, and a callback URL. Once activated, the standard (non-OIDC) cookie based lakeFS authentication will not be available. API access will still use credentials managed internally like today.
+
+As a first step, authorization will still be managed internally by lakeFS.
+
+### Open questions
+
+#### What do we save in our database?
+What information about users should we save in the lakeFS database?
+
+The minimum here is to save the unique user ID generated by the OIDC provider (the "subject" claim).
+
+However, this may not be enough. 
+Since we only want authentication (and not authorization) to be handled by the external provider, we should be able to link between a user and roles/policies. It seems that to do so we need to save some more info (at least username/email). Once we also move authorizaiton outside, it may be easier to avoid saving any PII on the lakeFS side.
+
+#### Can an installation deactivate OIDC after it's been turned on?
+
+If so, how does one migrate the existing users?
+
+#### How does one create the initial admin user?
```

Comment:
can't we use our current process that the first user is still required to provide the system?
if we support both internal authentication and oidc side by side, why does it matter?

Reply:
I was leaning towards not supporting them side-by-side, as this introduces some complications. 
Since I see there's other opinions, I added this as a question.